sql/core/src/test/scala/org/apache/spark/sql/execution/adaptive/AdaptiveQueryExecSuite.scala

```diff
@@ -1224,4 +1224,53 @@ class AdaptiveQueryExecSuite
       })
     }
   }
+
+  test("SPARK-32765: EliminateJoinToEmptyRelation should respect exchange behavior " +
+    "when canChangeNumPartitions == false") {
+    withSQLConf(
+      SQLConf.ADAPTIVE_EXECUTION_ENABLED.key -> "true",
+      // exclude ConvertToLocalRelation rule make it easier for Test.
+      SQLConf.OPTIMIZER_EXCLUDED_RULES.key -> ConvertToLocalRelation.ruleName) {
```

Comment:
We already exclude this in `SharedSparkSession`? https://github.com/apache/spark/blob/95f1e9549bb741db6c285390a71c609a9e5d3b02/sql/core/src/test/scala/org/apache/spark/sql/test/SharedSparkSession.scala#L70-L74